backend/src/db.py

```diff
@@ -0,0 +1,26 @@
+# db.py
+import os
+
+from dotenv import load_dotenv
+from supabase import Client, create_client
+
+# Load env variables from .env
+load_dotenv()
+
+SUPABASE_URL = os.getenv("SUPABASE_URL")
+SUPABASE_KEY = os.getenv("SUPABASE_KEY")
+
+
+def get_supabase_client() -> Client:
+    if not SUPABASE_URL or not SUPABASE_KEY:
+        raise Exception(
+            status_code=500, detail="Supabase credentials are not set"
```

Comment:
```suggestion
            status_code=500, detail="Supabase credentials missing in environment variables"
```

A little more specific.

Reply:
I would prefer more generic error messages to not leak any configuration details

Reply:
ok, convinced :)